x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/single_page_layout/index.tsx

```diff
@@ -267,11 +267,34 @@ export const CreatePackagePolicySinglePage: CreatePackagePolicyParams = ({
   );
 
   const extensionView = useUIExtension(packagePolicy.package?.name ?? '', 'package-policy-create');
+  const replaceDefineStepView = useUIExtension(
+    packagePolicy.package?.name ?? '',
+    'package-policy-replace-define-step'
+  );
+
+  if (replaceDefineStepView && extensionView) {
+    // eslint-disable-next-line no-console
+    console.warn(
+      "'package-policy-create' is ignored when 'package-policy-replace-define-step' is defined"
+    );
+  }
```

Comment:
not sure logging here is good enough

Reply:
Yeah I think `useUIExtensions` or adding an overload to `useUIExtension` that takes an array of extensions would be the path forward here. Agree that logging is not enough to prevent this, as we should prevent it entirely at compile time.

Reply:
@kpollich 

thought about this again and i believe we shouldn't overload `useUIExtension`, because:

 - registration is not really constrained, so we could narrow the types but that won't make them true
    - we can't fully limit registration with static types, as it can be done multiple times. 
    - we can make `registerExtension` take an array and type-check against invalid combinations, and run a full sweep on other plugins, which would somewhat mitigate possible future risk of misuse
 - there is actually not much to do with that information. we still need to account for both cases, and when doing those we don't need the other extension anyway. after making the types work, i noticed the distinction wasn't really used
 -  [overloading](https://github.com/elastic/kibana/commit/2f642c7cb2d21cca2f6a7d6ec9a80d5b47ba933a)  wasn't pretty

let me know what you think is best here/follow ups: 

 - add/iterate on overloading commit
 - just drop/keep the comments 
 - work on `registerExtension` 
 - something else?

 thanks! 

also note i'd still want to merge https://github.com/elastic/kibana/pull/149137 as soon as we can because there are other PRs depending on it

Reply:
Following up on our chat offline:

Let's just `throw` an error when we encounter this case. This should result in the Fleet UI's error boundary being displayed and should catch the "incompatible UI extension" state during development. I don't want to silently ignore a UI extension with a browser console log/warning because it will be non-obvious to someone developing in this situation. 

```ts
if (replaceDefineStepView && extensionView) { 
  throw new Error(
    "'package-policy-create' and 'package-policy-replace-define-step' cannot both be registered as UI extensions"
  )
}
```

Thanks for investigating here and following up.